askama_parser/src/expr.rs

```diff
@@ -72,6 +72,8 @@ pub enum Expr<'a> {
     Call(Box<Expr<'a>>, Vec<Expr<'a>>),
     RustMacro(Vec<&'a str>, &'a str),
     Try(Box<Expr<'a>>),
+    /// This variant should never be used directly. It is created when generating filter blocks.
+    Generated(String),
```

Comment:
That seems ugly. Why is it necessary?

Reply:
I need a variant with these two constraints:

 1. Doesn't take a reference (since it's a local variable).
 2. Doesn't make any transformation to it.

Reply:
I mean, why do we need store any generated stuff back into an `Expr`?

Reply:
Because we want to add the variable that was created to be used in the next `writer` use. I want this to be done in two steps: first we generate a `String` with the content of the filter and store it into a variable, then we use this variable alongside the rest of the template.